applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -230,41 +355,46 @@ static void pgps_event_handler(struct nrf_cloud_pgps_event *event)
 		break;
 	/* A P-GPS prediction is available now for the current date and time. */
 	case PGPS_EVT_AVAILABLE: {
+		LOG_INF("PGPS_EVT_AVAILABLE");
 		struct nrf_modem_gnss_agps_data_frame req;
 
-		LOG_INF("PGPS_EVT_AVAILABLE");
 		/* read out previous NRF_MODEM_GNSS_EVT_AGPS_REQ */
 		err = read_agps_req(&req);
 		if (err) {
-			/* Ephemerides assistance only */
-			LOG_INF("Failed to read A-GPS request: %d", err);
-			LOG_INF("Use ephemerides assistance only");
-			err = nrf_cloud_pgps_inject(event->prediction, NULL);
-		} else {
-			/* All assistance elements as requested */
-			err = nrf_cloud_pgps_inject(event->prediction, &req);
+			LOG_INF("Failed to read back A-GPS request (%d)."
+				" Using ephemerides assistance only.", err);
 		}
+		err = nrf_cloud_pgps_inject(event->prediction, err ? NULL : &req);
 		if (err) {
-			LOG_ERR("Unable to send prediction to modem: %d", err);
+			LOG_ERR("Failed to inject P-GPS data to modem (%d).", err);
 			break;
 		}
-		err = nrf_cloud_pgps_preemptive_updates();
-		if (err) {
-			LOG_ERR("Preemptive updates error: %d", err);
+		if (gnss_cloud_assistance) {
+			err = nrf_cloud_pgps_preemptive_updates();
+			if (err) {
+				/* This is expected to fail if new predictions are to be downloaded
+				 * but the connection to nRF Cloud is not available.
+				 * CONFIG_NRF_CLOUD_PGPS_REPLACEMENT_THRESHOLD controls
+				 * how often an attempt to download new predictions will be made.
+				 */
+				LOG_WRN("Failed to request new P-GPS predictions (%d).", err);
+			}
 		}
 	} break;
 	/* All P-GPS predictions are available. */
 	case PGPS_EVT_READY:
 		LOG_INF("PGPS_EVT_READY");
 		break;
-	default:
+	case PGPS_EVT_REQUEST:
```

Comment:
No default in switch-case always makes me suspicious that there should be something. Could we keep the default: break?

Reply:
It tends to increase readability of the code. In this case, I am reading the code and I am thinking, what will happen when a new event is added. Default: break, would indicate that we would ignore it. Logging unknown value would be even better.

That said, I did not know that compiler would generate an error in this case. Moreover, I did not even notice (and it is not obvious from the code) that the event->type is an enum, which would generate the said error.

So readability would have been the point. In this case though, there is also benefit for omitting the default as well, as that forces addition of new values.

Reply:
Can be resolved.